habitat/datasets/image_nav/instance_image_nav_dataset.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+import os
+from typing import Any, Dict, List, Optional
+
+from habitat.config import Config
+from habitat.core.registry import registry
+from habitat.core.simulator import AgentState
+from habitat.core.utils import DatasetFloatJSONEncoder
+from habitat.datasets.pointnav.pointnav_dataset import (
+    DEFAULT_SCENE_PATH_PREFIX,
+    PointNavDatasetV1,
+)
+from habitat.tasks.nav.instance_image_nav_task import (
+    InstanceImageGoal,
+    InstanceImageGoalNavEpisode,
+    InstanceImageParameters,
+)
+from habitat.tasks.nav.object_nav_task import ObjectViewLocation
+
+
+@registry.register_dataset(name="InstanceImageNav-v1")
+class InstanceImageNavDatasetV1(PointNavDatasetV1):
+    """Class that loads an Instance Image Navigation dataset."""
+
+    goals: Dict[str, InstanceImageGoal]
+    episodes: List[InstanceImageGoalNavEpisode] = []  # type: ignore
+
+    def __init__(self, config: Optional[Config] = None) -> None:
+        self.goals = {}
+        super().__init__(config)
+
+    def to_json(self) -> str:
+        for i in range(len(self.episodes)):
+            self.episodes[i].goals = []
```

Comment:
```suggestion
            self.episodes[i].goals.clear()
```
Makes it a bit more obvious you are clearing the list